src/dotty/tools/dotc/parsing/Parsers.scala

```diff
@@ -648,12 +648,18 @@ object Parsers {
     }
 
 /* ------------- TYPES ------------------------------------------------------ */
-    /** Same as [[typ]], but emits a syntax error if it returns a wildcard.
+    /** Same as [[typ]], but if this results in a wildcard it emits a syntax error and
+     *  returns a tree for type `Any` instead.
      */
     def toplevelTyp(): Tree = {
       val t = typ()
-      for (wildcardPos <- findWildcardType(t)) syntaxError("unbound wildcard type", wildcardPos)
-      t
+      findWildcardType(t) match {
+        case Some(wildcardPos) =>
+          syntaxError("unbound wildcard type", wildcardPos)
+          Select(Select(Ident(Names.termName("_root_")), Names.termName("scala")),
```

Comment:
instead of `Names.termName` and `Names.typeName` you should use the names already defined in [StdNames](https://github.com/lampepfl/dotty/blob/master/src/dotty/tools/dotc/core/StdNames.scala): `nme.ROOTPKG`, `nme.scala_` and `tpnme.Any`.